pkg/util/create_or_update.go

```diff
@@ -133,16 +134,27 @@ func CreateAnew(ctx context.Context, client resource.Interface, obj runtime.Obje
 	deleteOptions metav1.DeleteOptions) (runtime.Object, error) { // nolint:gocritic // Match K8s API
 	name := resource.ToMeta(obj).GetName()
 
-	var created runtime.Object
+	var retObj runtime.Object
 
 	err := wait.ExponentialBackoff(backOff, func() (bool, error) {
 		var err error
 
-		created, err = client.Create(ctx, obj, createOptions)
+		retObj, err = client.Create(ctx, obj, createOptions)
```

Comment:
If this returns with no error, can’t we assume that the created object is correct, and return early?

Reply:
That's the case on L144, ie `IsAlreadyExists` returns false if err is nil.

Reply:
Ah yes, my brain saw the error-handling and didn’t spot that it applied even if there was no error!